src/traces/box/attributes.js

```diff
@@ -62,6 +62,16 @@ module.exports = {
             'missing and the position axis is categorical'
         ].join(' ')
     },
+    text: extendFlat({}, scatterAttrs.text, {
```

Comment:
Text is expected to be an array as long as the sample array `y` (`x`) for vertical (horizontal) box plots, matching each sample pt one-to-one.